### ISSUE TITLE
GNOME 3 shutdown is delayed for 2 minutes: “A stop job is running for User Manager for UID 1000”

### ISSUE THREAD

**Issue by andersk** [NixOS/nixpkgs#113165]
**Describe the bug**

In `nixos-unstable` with `services.xserver.desktopManager.gnome3.enable = true`, shutdown is delayed for 2 minutes at:

```
[ ***  ] A stop job is running for User Manager for UID 1000 (1min 54s / 2min 1s)
```

**To Reproduce**
This can be reproduced with a minimal VM. (It also happens with a real system, of course.)

1. Create `/tmp/configuration.nix`:
   ```nix
   {
     services.xserver.enable = true;
     services.xserver.desktopManager.gnome3.enable = true;
     users.users.nixos.isNormalUser = true;
     users.users.nixos.hashedPassword = "";
   }
   ```
2. Checkout `nixos-unstable` (currently ff96a0fa5635770390b184ae74debea75c3fd534).
2. Run: `nix-shell -I nixos-config=/tmp/configuration.nix -p '(import ./nixos {}).vm' --run 'run-nixos-vm -m 4096'`
3. Click Log In.
4. Click the GNOME power menu and select Power Off; click Power Off.
5. The shutdown is delayed for 2 minutes:
   ```
   [ ***  ] A stop job is running for User Manager for UID 1000 (1min 54s / 2min 1s)
   ```

**Expected behavior**
Shutdown should not be delayed.

**Additional context**
This was not an issue in `nixos-20.09`.

**Notify maintainers**
@hedning @jtojnar @worldofpeace @dasj19

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.15, NixOS, 21.05.git.ff96a0fa563 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels(anders): `""`
 - channels(root): `"nixos-21.05pre269929.ff96a0fa563"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: []
# a list of nixos modules affected by the problem
module: [services.xserver.desktopManager.gnome3]
```

**Comment by worldofpeace** [comment]
I can reproduce, indeed very annoying. A workaround can be to reduce the systemd timeout time. I've heard other people report this

**Comment by andersk** [comment]
I found the issue and fix from upstream: #113171.
